### PR TITLE
Adding parser

### PR DESCRIPTION
**Pull request by repaj**
## Purpose
This language needs basic tools to text analysing, that is, parser.

## Approach
This is basic implementation of ANTLR 4 parser with simple facade that generates AST tree
using composite pattern. AST tree could be analysed using visitor pattern.

## Learning
[ANTLR 4 Tutorial](https://tomassetti.me/antlr-mega-tutorial/)
[Composite Pattern](https://en.wikipedia.org/wiki/Composite_pattern)
[Visitor Pattern](https://en.wikipedia.org/wiki/Visitor_pattern)

### CODE REVIEW

**Comment by repaj** [comment]
There is missing unit tests for parser grammar.

File: src/main/antlr/LpToJvm.g4 [hunk 1]
```diff
@@ -0,0 +1,37 @@
+grammar LpToJvm;
+
+compilationUnit : lambdaTerm EOF;
+
+lambdaTerm  : '(' lambdaTerm ')'                        #parenTerm
+            | literal                                   #literalTerm
+            | ID                                        #identifierTerm
+            | abstraction                               #abstractionTerm
+            | lambdaTerm lambdaTerm                     #leftApplicationTerm
+            |<assoc=right> lambdaTerm '$' lambdaTerm    #rightApplicationTerm
```
Comment (KatJon) [inline src/main/antlr/LpToJvm.g4]: missing space

File: src/main/antlr/LpToJvm.g4 [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+fragment PLUS_MINUS : ('+' | '-')? ;
+FLOAT
+    :   PLUS_MINUS INT '.' INT EXP?   // 1.35, 1.35E-9, 0.3, -4.5
+    |   PLUS_MINUS INT EXP            // 1e10 -3e4
+    |   PLUS_MINUS INT                // -3, 45
+    ;
+INT :   PLUS_MINUS ('0' | [1-9] [0-9]*) ; // no leading zeros
+fragment EXP :   [Ee] [+\-]? INT ;
+STRING : '"' (~'"'|'\\"')* '"'  ;
+BOOLEAN : 'true' | 'false';
```
Comment (KatJon) [inline src/main/antlr/LpToJvm.g4]: Did you check, that it recognizes boolean and not treat it as a placeholder?

Comment (KatJon) [reply src/main/antlr/LpToJvm.g4]: Maybe allow only the lowercase placeholders, and the constants will be uppercase

Comment (repaj) [reply src/main/antlr/LpToJvm.g4]: I'll push this feature one more time with unit test.

File: src/main/antlr/LpToJvm.g4 (hunk 1 above)
Comment (repaj) [reply src/main/antlr/LpToJvm.g4]: It's fixed in ee52f33.

File: src/main/antlr/LpToJvm.g4 (hunk 2 above)
Comment (repaj) [reply src/main/antlr/LpToJvm.g4]: See 15499b1.

**Pull request merged**